zfit/core/limits.py

```diff
@@ -0,0 +1,476 @@
+from __future__ import print_function, division, absolute_import
+
+import inspect
+from typing import Tuple, Union
+
+import numpy as np
+
+from zfit.util.exception import NormRangeNotImplementedError, MultipleLimitsNotImplementedError
+
+
+class Range(object):
+    FULL = object()  # unique reference
+    ANY = object()
+    ANY_LOWER = object()  # TODO: need different upper, lower?
+    ANY_UPPER = object()
+    __HASH_DELIMINATOR = object()
+
+    def __init__(self, *, limits=None, lower=None, upper=None, dims=None,
+                 convert_none=False):  # TODO: change lower, upper?
+        """Range holds limits and specifies dimension.
+
+        Args:
+
```

Comment:
Missing Args

Reply:
Done (will be fully done when the docs automation is set up)